pkg/asset/installconfig/aws/validation.go

```diff
@@ -266,6 +269,12 @@ func validateMachinePool(ctx context.Context, meta *Metadata, fldPath *field.Pat
 func validateSecurityGroupIDs(ctx context.Context, meta *Metadata, fldPath *field.Path, platform *awstypes.Platform, pool *awstypes.MachinePool) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	// The installer also creates a security group: `${var.cluster_id}-master-sg/${var.cluster_id}-worker-sg`
+	if len(pool.AdditionalSecurityGroupIDs) > maxSecurityGroupsCount-1 {
```

Comment:
this validation is static (does not depend on api calls) so it could and probably should go in pkg types.

Reply:
That makes so much more sense.